File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/messages-thinking.py
import os

from anthropic import Anthropic

from common import finish, start


start("anthropic")
client = Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"], base_url=os.environ["PROVIDER_BASE_URL"])
client.messages.create(
    model="claude-3-7-sonnet-20250219",
    max_tokens=32,
    thinking={"type": "enabled", "budget_tokens": 1024},
    messages=[{"role": "user", "content": "Say parity."}],
)
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/messages-thinking.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('anthropic')
  const { Anthropic } = loadVersionedModule('@anthropic-ai/sdk')
  const client = new Anthropic({ apiKey: process.env.ANTHROPIC_API_KEY, baseURL: process.env.PROVIDER_BASE_URL })
  await client.messages.create({
    model: 'claude-3-7-sonnet-20250219',
    max_tokens: 32,
    thinking: { type: 'enabled', budget_tokens: 1024 },
    messages: [{ role: 'user', content: 'Say parity.' }],
  })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})
